mlflow/server/auth/__init__.py

```diff
@@ -55,6 +137,226 @@ def make_basic_auth_response() -> Response:
+    )
+
+
+def _get_permission_from_run_id() -> Permission:
+    # run permissions inherit from parent resource (experiment)
+    # so we just get the experiment permission
+    run_id = _get_request_param("run_id")
+    run = get_run(run_id)
+    experiment_id = run.info.experiment_id
+    username = request.authorization.username
+    return _get_permission_from_store_or_default(
+        lambda: store.get_experiment_permission(experiment_id, username).permission
+    )
+
+
+def _get_permission_from_registered_model_name() -> Permission:
+    name = _get_request_param("name")
+    username = request.authorization.username
+    return _get_permission_from_store_or_default(
+        lambda: store.get_registered_model_permission(name, username).permission
+    )
+
+
+def validate_can_read_experiment():
+    return _get_permission_from_experiment_id().can_read
+
+
+def validate_can_update_experiment():
+    return _get_permission_from_experiment_id().can_update
+
+
+def validate_can_delete_experiment():
+    return _get_permission_from_experiment_id().can_delete
+
+
+def validate_can_manage_experiment():
+    return _get_permission_from_experiment_id().can_manage
+
+
+def validate_can_read_run():
+    return _get_permission_from_run_id().can_read
+
+
+def validate_can_update_run():
+    return _get_permission_from_run_id().can_update
+
+
+def validate_can_delete_run():
+    return _get_permission_from_run_id().can_delete
+
+
+def validate_can_manage_run():
+    return _get_permission_from_run_id().can_manage
+
+
+def validate_can_read_registered_model():
+    return _get_permission_from_registered_model_name().can_read
+
+
+def validate_can_update_registered_model():
+    return _get_permission_from_registered_model_name().can_update
+
+
+def validate_can_delete_registered_model():
+    return _get_permission_from_registered_model_name().can_delete
+
+
+def validate_can_manage_registered_model():
+    return _get_permission_from_registered_model_name().can_manage
+
+
+def sender_is_admin():
+    """Validate if the sender is admin"""
+    username = request.authorization.username
+    return store.get_user(username).is_admin
+
+
+def username_is_sender():
+    """Validate if the request username is the sender"""
+    username = _get_request_param("username")
+    sender = request.authorization.username
+    return username == sender
+
+
+def validate_can_read_user():
+    return username_is_sender()
+
+
+def validate_can_update_user_password():
+    return username_is_sender()
+
+
+def validate_can_update_user_admin():
+    # only admins can update, but admins won't reach this validator
+    return False
+
+
+def validate_can_delete_user():
+    # only admins can delete, but admins won't reach this validator
+    return False
+
+
+BEFORE_REQUEST_HANDLERS = {
+    # Routes for experiments
+    GetExperiment: validate_can_read_experiment,
+    GetExperimentByName: validate_can_read_experiment,
```

Comment:
It looks like this results in an error because the `/experiments/get-by-name` doesn't have a `experiment_id` parameter:

```
2023/04/24 11:26:07 ERROR mlflow.server: Exception on /ajax-api/2.0/mlflow/experiments/get-by-name [GET]
Traceback (most recent call last):
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 1821, in full_dispatch_request
    rv = self.preprocess_request()
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 2312, in preprocess_request
    rv = self.ensure_sync(before_func)()
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 385, in _before_request
    if not validator():
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 210, in validate_can_read_experiment
    return _get_permission_from_experiment_id().can_read
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 182, in _get_permission_from_experiment_id
    experiment_id = _get_request_param("experiment_id")
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 158, in _get_request_param
    raise MlflowException(
mlflow.exceptions.MlflowException: Missing value for required parameter 'experiment_id'. See the API docs for more information about request parameters.
```